app/reducers/app.ts

```diff
@@ -901,9 +901,12 @@ export const actions = {
     function loadDirectoryContentInt(fsEntryMeta?: FileSystemEntryMeta) {
       // Uncomment the following line will to clear all content before loading new dir content
       dispatch(actions.loadDirectorySuccessInt(directoryPath, [], true)); // this is to reset directoryContent (it will reset color too)
-      const location = extractLocation(directoryPath, locations);
-      if (location !== undefined) {
-        dispatch(actions.changeLocation(location));
+      if (directoryPath) {
```

Comment:
Without this check if you have 2 S3 Locations and click on the second one (make it CurrentLocation). After you click in subfolder in the current selected Location another one on the top position will be selected.

Reply:
top!